tests/helpers/environment.ts

```diff
@@ -0,0 +1,128 @@
+import path from 'path';
```

Comment:
I've tried to leave comments describing what each thing does, but here is a high level:

* To set up a test, we need to create two repos. We will use one repo to run the action script, but we also need another repo that the first one can talk to so that the action script can run `git fetch --tags`. 
* The collection of these repos is called an environment and are kept in a directory within a temporary location. The environment is cleared and recreated before each test.
* There are two types of Environment classes, one which is specialized to initialize a polyrepo and one to initialize a monorepo. These have different characteristics, so there are two kinds of Repo classes as well.